File: app/src/main/java/example/prada/lab/pradaoutlook/store/BaseEventStore.java
package example.prada.lab.pradaoutlook.store;

import java.util.ArrayList;
import java.util.List;

import example.prada.lab.pradaoutlook.model.IEventDataUpdatedListener;

/**
 * Created by prada on 11/1/16.
 */

public abstract class BaseEventStore implements IEventStore {

    List<IEventDataUpdatedListener> mListeners = new ArrayList<>();

    @Override
    public final void addListener(IEventDataUpdatedListener listener) {
        if (listener == null || mListeners.contains(listener)) {
            return;
        }
        mListeners.add(listener);
    }

    @Override
    public final void removeListener(IEventDataUpdatedListener listener) {
        if (listener == null || mListeners.contains(listener)) {
            return;
        }
        mListeners.remove(listener);
    }
}

File: app/src/main/java/example/prada/lab/pradaoutlook/store/BaseEventStore.kt
package example.prada.lab.pradaoutlook.store

import example.prada.lab.pradaoutlook.model.IEventDataUpdatedListener

abstract class BaseEventStore : IEventStore {
    @JvmField protected val mListeners = ArrayList<IEventDataUpdatedListener>()

    override fun addListener(listener: IEventDataUpdatedListener?) {
        if (listener != null && !mListeners.contains(listener)) mListeners.add(listener)
    }

    override fun removeListener(listener: IEventDataUpdatedListener?) {
        if (listener != null) mListeners.remove(listener)
    }
}
